crawlfrontier/contrib/scrapy/schedulers/frontier.py

```diff
@@ -154,4 +154,12 @@ def _get_exception_code(self, exception):
             return '?'
 
     def _request_is_redirected(self, request):
-        return request.meta.get('redirect_times', 0) > 0
\ No newline at end of file
+        return request.meta.get('redirect_times', 0) > 0
+
+    def _get_overused_keys(self):
+        keys = []
+        downloader = self.crawler.engine.downloader
+        for (key, slot) in downloader.slots.iteritems():
+            if len(slot.active) / float(slot.concurrency) > 1.0:
```

Comment:
I am not sure this is the right formula to use here.

How about we use a scrapy setting which specifies the maximum queued requests per key (domain/IP) before we consider it overused? I think that it's a good idea to allow some queued requests, since the current requests may finish before the frontier's get_next_requests is called again.